aim/db/migration/alembic_migrations/versions/5e29ae4f45e6_host_domain_add_primary_key.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2017 Cisco Systems
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""Convert vmm_domain_name to primary key in HostDomainMapping
+
+Revision ID: 5e29ae4f45e6
+Revises: 593d228d2fb4
+Create Date: 2017-10-25 16:39:33.713695
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '5e29ae4f45e6'
+down_revision = '593d228d2fb4'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+AIM_HOST_DOMAIN_MAPPING = 'aim_host_domain_mapping'
+
+# A model of the networks table with the fields present before
+# the migration.
+domains = sa.Table(AIM_HOST_DOMAIN_MAPPING, sa.MetaData(),
+                   sa.Column('host_name', sa.String(128), primary_key=True),
+                   sa.Column('vmm_domain_name', sa.String(64)),
+                   sa.Column('physical_domain_name', sa.String(64)))
+
+
+def upgrade():
+    session = sa.orm.Session(bind=op.get_bind())
+    additions = get_values(session, domains)
+
+    op.drop_table(AIM_HOST_DOMAIN_MAPPING)
+    op.create_table(
+        AIM_HOST_DOMAIN_MAPPING,
+        sa.Column('host_name', sa.String(128)),
+        sa.Column('domain_name', sa.String(64)),
+        sa.Column('domain_type', sa.Enum('PhysDom',
+                                         'OpenStack',
+                                         'Kubernetes',
+                                         'VMware')),
+        sa.PrimaryKeyConstraint('host_name', 'domain_name')
```

Comment:
all 3 fields should be the primary key I think.

Reply:
Will fix (as above).